### PR TITLE
FEATURE: Set topic as Q&A formatted using Topic#subtype.

### PR DESCRIPTION
**Pull request by tgxworld**
Before this change, we were depending on setting a special tag or
enabling the Q&A format at the category level. However, we want to
simplify this by simply making this a topic level setting.

### Screenshots

Peek 2022-03-21 15-52

### CODE REVIEW

File: test/javascripts/acceptance/composer-test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    await composerActionTitle.expand();
+    await composerActionTitle.selectKitSelectRowByName(
+      I18n.t("composer.composer_actions.remove_as_qa.label")
+    );
+
+    assert.notStrictEqual(
+      query(".action-title").textContent.trim(),
+      I18n.t("composer.create_qa.label"),
+      "revets to original composer title when Q&A format is disabled"
```
Comment (CvX) [inline test/javascripts/acceptance/composer-test.js:62]:
```suggestion
      "reverts to original composer title when Q&A format is disabled"
```

File: spec/models/topic_spec.rb [hunk 2]
```diff
@@ -5,30 +5,54 @@
...
+  let(:up) { QuestionAnswerVote.directions[:up] }
 
-  before do
-    SiteSetting.qa_tags = tag.name
-    topic.tags << tag
-    comments
-  end
+  context "validations" do
+    describe '#subtype' do
+      it "should not allow Q&A formated topics to be created when qa_enabled site setting is not enabled" do
```
Comment (CvX) [inline spec/models/topic_spec.rb:27]:
```suggestion
      it "should not allow Q&A formatted topics to be created when qa_enabled site setting is not enabled" do
```

**Review by CvX — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you @CvX

**Comment by pmusaraj** [comment]
@tgxworld with this change, we need to expose a way to toggle the Q&A status of an existing topic. Previously you would do this by adding/removing the tag. But in the editor, even if I edit the OP, I don't see a way to toggle Q&A from the same button.

**Comment by tgxworld** [comment]
@pmusaraj In this case, I actually added a validation to prevent people from updating a topic to be Q&A formatted. The reason I chose to do so is because the format will not make sense if you try and convert random topics to be Q&A formatted. For example, `Post` with `reply_to_post_number != 1` are not allowed. Perhaps we only allow people to update if `Topic#post_count == 1`?

**Comment by pmusaraj** [comment]
@tgxworld yes, I think we should at least allow toggling QA status of a topic as long as it has no replies, in fact, I'd go as far as allowing toggling as long as it has no replies to replies. 

More generally, I also wonder whether we should conform to the pattern used by the solved/voting plugins, where the feature is enabled per category. Might be worth discussing this internally.

**Pull request merged**